### ISSUE TITLE
Arr::get() should return $default if $key is null, not $array

### ISSUE THREAD

**Issue by duartefq** [laravel/ideas#1315]
This would be more fitting to the method's purpose IMO. Currently, `Arr::get()`[ returns the whole array](https://github.com/illuminate/support/blob/master/Arr.php#L285) if `$key` is null.

**Comment by michaeldyrynda** [comment]
Seems like a significant breaking change. Why are you passing `null` to the method? Perhaps docs should clarify that it will only return the default if the key does not exist.

**Comment by HongjiangHuang** [comment]
This change will affect many project

**Comment by dbpolito** [comment]
I agree with this ticket, not sure if it will break many applications as it's a very odd behaviour.

Think about this use case:

```
$typeId = array_get($types, $request->type, 'something')
```

If an input or query string is optional and you want to map something it's very easy to break.

In my use case i just casted to string to workaround this issue:

```
$typeId = array_get($types, (string) $request->type, 'something')
```

**Comment by davidvandertuijn** [comment]
null can be passed unintentionally, instead of the default value an Array is returned, and this breaks my code.
